test/integration/stats_test.lua

```diff
@@ -0,0 +1,586 @@
+-- spaces and ensure statistics is updated.
+for name, case in pairs(simple_operation_cases) do
+    local test_name = ('test_%s'):format(name)
+
+    if case.prepare ~= nil then
+        g.before_test(test_name, case.prepare)
+    end
+
+    g[test_name] = function(g)
+        -- Collect stats before call.
+        local stats_before = g:get_stats(space_name)
+        t.assert_type(stats_before, 'table')
+
+        -- Call operation.
+        local before_start = clock.monotonic()
+
+        local _, err
+        if case.eval ~= nil then
+            if case.pcall then
+                _, err = pcall(g.router.eval, g.router, case.eval, case.args)
+            else
+                _, err = g.router:eval(case.eval, case.args)
+            end
+        else
+            _, err = g.router:call(case.func, case.args)
+        end
+
+        local after_finish = clock.monotonic()
+
+        if case.expect_error ~= true then
+            t.assert_equals(err, nil)
+        else
+            t.assert_not_equals(err, nil)
+        end
+
+        -- Collect stats after call.
+        local stats_after = g:get_stats(space_name)
+        t.assert_type(stats_after, 'table')
+        t.assert_not_equals(stats_after[case.op], nil)
+
+        -- Expecting 'ok' metrics to change on `expect_error == false`
+        -- or 'error' to change otherwise.
+        local changed, unchanged
+        if case.expect_error == true then
+            changed = 'error'
+            unchanged = 'ok'
+        else
+            unchanged = 'error'
+            changed = 'ok'
+        end
+
+        local op_before = set_defaults_if_empty(stats_before, case.op)
+        local changed_before = op_before[changed]
+        local op_after = set_defaults_if_empty(stats_after, case.op)
+        local changed_after = op_after[changed]
+
+        t.assert_equals(changed_after.count - changed_before.count, 1,
+            'Expected count incremented')
+
+        local ok_latency_max = math.max(changed_before.latency, after_finish - before_start)
+
+        t.assert_gt(changed_after.latency, 0,
+            'Changed latency has appropriate value')
+        t.assert_le(changed_after.latency, ok_latency_max,
+            'Changed latency has appropriate value')
+
+        local time_diff = changed_after.time - changed_before.time
+
+        t.assert_gt(time_diff, 0, 'Total time increase has appropriate value')
+        t.assert_le(time_diff, after_finish - before_start,
+            'Total time increase has appropriate value')
+
+        local unchanged_before = op_before[unchanged]
+        local unchanged_after = stats_after[case.op][unchanged]
+
+        t.assert_equals(unchanged_before, unchanged_after, 'Other stats remained the same')
+    end
+end
+
+
+-- Call some non-select operations for non-existing
+-- spaces and ensure statistics is updated.
+for name, case in pairs(unknown_space_cases) do
+    local test_name = ('test_%s_on_unknown_space'):format(name)
```

Comment:
But `unknown_space_cases` has select and pairs cases. Am I missed something?

Reply:
I'll check it and fix (as part of this PR)

Reply:
I don't see this code now, so I consider it resolved.